build/dnvm.ps1

```diff
@@ -177,11 +184,24 @@ if($CmdPathFile) {
 # Determine where runtimes can exist (RuntimeHomes)
 if(!$RuntimeHomes) {
     # Set up a default value for the runtime home
-    $UnencodedHomes = "%USERPROFILE%\$DefaultUserDirectoryName"
+    $UnencodedHomes = "$env:USERPROFILE\$DefaultUserDirectoryName;$GlobalHome\$DefaultGlobalDirectoryName"
 } else {
     $UnencodedHomes = $RuntimeHomes
 }
 
+# Determine the default global installation directory (GlobalHome)
+if(!$GlobalHome) {
+    if($env:ProgramData) {
+        $GlobalHome = "$env:ProgramData\$DefaultGlobalDirectoryName"
+    } else {
+        $GlobalHome = "$env:AllUsersProfile\$DefaultGlobalDirectoryName"
+    }
+
+    $env:DNX_GLOBAL_HOME="$GlobalHome"
+    $env:DNX_HOME="$env:DNX_HOME;$env:DNX_GLOBAL_HOME"
+    $UnencodedHomes = "$UnencodedHomes;$GlobalHome"
+}
+
 $UnencodedHomes = $UnencodedHomes.Split(";")
 $RuntimeHomes = $UnencodedHomes | ForEach-Object { [Environment]::ExpandEnvironmentVariables($_) }
 $RuntimeDirs = $RuntimeHomes | ForEach-Object { Join-Path $_ "runtimes" }
```

Comment:
Until we fix aspnet/Home#899 and / or aspnet/dnvm#422, we should not take this version of `dnvm.ps1`. The problem is that all variables in `$UnencodedHomes` are already expanded when `[Environment]::ExpandEnvironmentVariables()` is called above i.e. they're referenced before they are definitely defined -- `$env:DNX_HOME` and `$env:DNX_GLOBAL_HOME` in particular. That leads to failures because `$_` is empty here.